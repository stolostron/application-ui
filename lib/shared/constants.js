/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2019. All Rights Reserved.
 * Copyright (c) 2021 Red Hat, Inc.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
'use strict'

module.exports.ACM_ACCESS_COOKIE = 'acm-access-token-cookie'
module.exports.ACM_AUTH_COOKIE = 'acm-acs-auth-cookie'
module.exports.ACM_USER_COOKIE = 'acm-user-cookie'
module.exports.ACM_REFRESH_INTERVAL_COOKIE = 'acm-refresh-interval-cookie'

module.exports.MCM_DIAGRAM_FILTER_COOKIE = 'mcm-diagram-filter-cookie'
module.exports.MCM_DESIGN_SPLITTER_OPEN_COOKIE =
  'mcm-design-splitter-open-cookie'
module.exports.MCM_DESIGN_SPLITTER_SIZE_COOKIE =
  'mcm-design-splitter-size-cookie'
module.exports.OVERVIEW_REFRESH_INTERVAL_COOKIE =
  'mcm-overview-interval-refresh-cookie'
module.exports.DIAGRAM_REFRESH_INTERVAL_COOKIE =
  'mcm-diagram-interval-refresh-cookie'
module.exports.REFRESH_TIMES = [15, 30, 60, 5 * 60, 30 * 60, 0]

module.exports.OVERVIEW_QUERY_COOKIE = 'mcm-overview-query-cookie'
module.exports.OVERVIEW_STATE_COOKIE = 'mcm-overview-state-cookie'
module.exports.DIAGRAM_QUERY_COOKIE = 'mcm-diagram-query-cookie'

module.exports.MAX_CHART_DATA_SIZE = 30

module.exports.LOCAL_HUB_NAME = 'local-cluster'

module.exports.RESOURCE_TYPES = {
  QUERY_APPLICATIONS: {
    name: 'QueryApplications',
    list: 'QueryApplicationList'
  },
  QUERY_SUBSCRIPTIONS: {
    name: 'QuerySubscriptions',
    list: 'QuerySubscriptionList'
  },
  QUERY_PLACEMENTRULES: {
    name: 'QueryPlacementRules',
    list: 'QueryPlacementRuleList'
  },
  QUERY_CHANNELS: {
    name: 'QueryChannels',
    list: 'QueryChannelList'
  },
  HCM_APPLICATIONS: { name: 'HCMApplication', list: 'HCMApplicationList' },
  HCM_CHANNELS: { name: 'HCMChannel', list: 'HCMChannelList' },
  HCM_SUBSCRIPTIONS: { name: 'HCMSubscription', list: 'HCMSubscriptionList' },
  HCM_PLACEMENT_RULES: {
    name: 'HCMPlacementRule',
    list: 'HCMPlacementRuleList'
  },
  HCM_FILTER_LIST: { name: 'HCMFilterList', list: 'HCMFilterList' },
  HCM_TOPOLOGY: { name: 'HCMTopology', list: 'HCMTopology' },
  USER_INFO: { name: 'userInfo', list: 'userInfoList' }
}

module.exports.MCM_CLUSTERS_SERVICES_ACTIONS = {}

module.exports.DOC_PATH_BASE =
  'https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.2/html/manage_applications/managing-applications'

module.exports.DOC_LINKS = {
  HOME: `${module.exports.DOC_PATH_BASE}`,
  TERMINOLOGY: `${
    module.exports.DOC_PATH_BASE
  }#application-model-and-definitions`,
  APPLICATIONS: `${
    module.exports.DOC_PATH_BASE
  }#managing-application-resources`,
  SUBSCRIPTIONS: `${
    module.exports.DOC_PATH_BASE
  }#creating-and-managing-subscriptions`,
  PLACEMENT_RULES: `${
    module.exports.DOC_PATH_BASE
  }#creating-and-managing-placement-rules`,
  CHANNELS: `${module.exports.DOC_PATH_BASE}#creating-and-managing-channels`
}
