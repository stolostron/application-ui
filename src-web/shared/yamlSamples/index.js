/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 * Copyright (c) 2020 Red Hat, Inc.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
import msgs from '../../../nls/platform.properties'

export const applicationSample =
  'apiVersion: v1______________________createApplication-namespace-apiVersion\nkind: Namespace\nmetadata:\n  name: ____________________________createApplication-namespace-metadata-name\n---\napiVersion: app.k8s.io/v1beta1\nkind: Application\nmetadata:\n  name: __________________________createApplication-metadata-name\n  namespace: _____________________createApplication-metadata-namespace\nspec:\n  componentKinds:\n  - group: apps.open-cluster-management.io\n    kind: Subscription\n  descriptor: {}\n  selector:\n    matchExpressions:\n      - key: _____________________createApplication-spec-selector-matchExpressions-key\n        operator: In\n        values: __________________createApplication-spec-selector-matchExpressions-key-values'
export const channelNamespaceSample =
  'apiVersion: v1______________________createChannel-namespace-apiVersion\nkind: Namespace\nmetadata:\n  name: ____________________________createChannel-namespace-metadata-name\n---\napiVersion: apps.open-cluster-management.io/v1\nkind: Channel\nmetadata:\n  name: ____________________________createChannel-metadata-name\n  namespace: _______________________createChannel-metadata-namespace\nspec:\n  gates:\n    annotations: ___________________createChannel-specNamespace-gates-annotations\n  pathname: foo ____________________createChannel-specNamespace-pathname\n  sourceNamespaces: ________________createChannel-specNamespace-sourceNamespaces\n  type: Namespace'
export const channelHelmRepoSample =
  'apiVersion: v1______________________createChannel-namespace-apiVersion\nkind: Namespace\nmetadata:\n  name: ____________________________createChannel-namespace-metadata-name\n---\napiVersion: apps.open-cluster-management.io/v1\nkind: Channel\nmetadata:\n  name: ____________________________createChannel-metadata-name\n  namespace: _______________________createChannel-metadata-namespace\nspec:\n  pathname: ________________________createChannel-specHelmRepo-pathname\n  configRef:\n    name: __________________________createChannel-specHelmRepo-configRef-name\n  type: HelmRepo\n---\napiVersion: v1______________________createChannel-configMap-apiVersion\nkind: ConfigMap\nmetadata:\n  name: ____________________________createChannel-configMap-metadata-name\n  namespace: _______________________createChannel-configMap-metadata-namespace'
export const channelObjectBucketSample =
  'apiVersion: v1______________________createChannel-namespace-apiVersion\nkind: Namespace\nmetadata:\n  name: ____________________________createChannel-namespace-metadata-name\n---\napiVersion: apps.open-cluster-management.io/v1\nkind: Channel\nmetadata:\n name: _____________________________createChannel-metadata-name\n namespace: ________________________createChannel-metadata-namespace\nspec:\n type: ObjectBucket\n pathname: _________________________createChannel-specObjectBucket-pathname\n secretRef:\n   name: ___________________________createChannel-specObjectBucket-secretRef-name\n gates:\n   annotations: ____________________createChannel-specObjectBucket-gate-annotations'
export const channelGitRepoSample =
  'apiVersion: v1______________________createChannel-namespace-apiVersion\nkind: Namespace\nmetadata:\n  name: ____________________________createChannel-metadata-name\n---\napiVersion: apps.open-cluster-management.io/v1\nkind: Channel\nmetadata:\n  name: ____________________________createChannel-metadata-name\n  namespace: _______________________createChannel-metadata-namespace\nspec:\n  type: GitHub\n  pathname: ________________________createChannel-specGitRepo-pathname'
export const subscriptionSample =
  'apiVersion: v1______________________createSubscription-namespace-resource\nkind: Namespace\nmetadata:\n  name: ____________________________createSubscription-name-resource\n---\napiVersion: apps.open-cluster-management.io/v1\nkind: Subscription\nmetadata:\n  name: ____________________________createSubscription-metadata-name\n  namespace: _______________________createSubscription-metadata-namespace\nspec:\n  channel: _________________________createSubscription-spec-channel\n  placement: \n    placementRef: \n      kind: PlacementRule\n      name: ________________________createSubscription-spec-placement-placementRef-name\ntimeWindow: \n  type: \'"active"\'\n  location: \'"America/Toronto"\'\n  weekdays: ________________________createSubscription-timeWindow-weekdays\n  hours: ___________________________createSubscription-timeWindow-hours'
export const placementRuleSample =
  'apiVersion: apps.open-cluster-management.io/v1\nkind: PlacementRule\nmetadata:\n  name: ____________________________createPlacementRule-metadata-name\n  namespace: _______________________createPlacementRule-metadata-namespace\nspec:\n  clusterSelector:\n    matchLabels: ___________________createPlacementRule-spec-clusterSelector-matchLabels\n  clusterReplicas: _________________createPlacementRule-spec-clusterReplicas'

export const getChannelSampleByType = channelSampleType => {
  let channelSample = ''
  if (channelSampleType === 'Namespace') {
    channelSample = channelNamespaceSample
  } else if (channelSampleType === 'HelmRepo') {
    channelSample = channelHelmRepoSample
  } else if (channelSampleType === 'ObjectBucket') {
    channelSample = channelObjectBucketSample
  } else if (channelSampleType === 'GitRepo') {
    channelSample = channelGitRepoSample
  }

  return channelSample
}

export const getChannelSample = (channelSampleType, locale) => {
  const channelSample = getChannelSampleByType(channelSampleType)
  const mapObj = {
    'createChannel-namespace-apiVersion':
      '# ' + msgs.get('description.createChannel.namespace.apiVersion', locale),
    'createChannel-namespace-metadata-name':
      '# ' +
      msgs.get('description.createChannel.namespace.metadata.name', locale),
    'createChannel-metadata-name':
      '# ' + msgs.get('description.createChannel.metadata.name', locale),
    'createChannel-metadata-namespace':
      '# ' + msgs.get('description.createChannel.metadata.namespace', locale),
    'createChannel-specNamespace-gates-annotations':
      '# ' +
      msgs.get(
        'description.createChannel.specNamespace.gates.annotations',
        locale
      ),
    'createChannel-specNamespace-pathname':
      '# ' +
      msgs.get('description.createChannel.specNamespace.pathname', locale),
    'createChannel-specNamespace-sourceNamespaces':
      '# ' +
      msgs.get(
        'description.createChannel.specNamespace.sourceNamespaces',
        locale
      ),
    'createChannel-specHelmRepo-pathname':
      '# ' +
      msgs.get('description.createChannel.specHelmRepo.pathname', locale),
    'createChannel-specHelmRepo-configRef-name':
      '# ' +
      msgs.get('description.createChannel.specHelmRepo.configRef.name', locale),
    'createChannel-specHelmRepo-type':
      '# ' + msgs.get('description.createChannel.specHelmRepo.type', locale),
    'createChannel-configMap-apiVersion':
      '# ' + msgs.get('description.createChannel.configMap.apiVersion', locale),
    'createChannel-configMap-metadata-name':
      '# ' +
      msgs.get('description.createChannel.configMap.metadata.name', locale),
    'createChannel-configMap-metadata-namespace':
      '# ' +
      msgs.get(
        'description.createChannel.configMap.metadata.namespace',
        locale
      ),
    'createChannel-specObjectBucket-pathname':
      '# ' +
      msgs.get('description.createChannel.specObjectBucket.pathname', locale),
    'createChannel-specObjectBucket-secretRef-name':
      '# ' +
      msgs.get(
        'description.createChannel.specObjectBucket.secretRef.name',
        locale
      ),
    'createChannel-specObjectBucket-gate-annotations':
      '# ' +
      msgs.get(
        'description.createChannel.specObjectBucket.gates.annotations',
        locale
      ),
    'createChannel-specGitRepo-pathname':
      '# ' + msgs.get('description.createChannel.specGitRepo.pathname', locale),
    null: '',
    _: ' '
  }

  var sample = channelSample.replace(
    /createChannel-namespace-apiVersion|createChannel-namespace-metadata-name|createChannel-metadata-namespace|createChannel-metadata-name|createChannel-specNamespace-gates-annotations|createChannel-specNamespace-pathname|createChannel-specNamespace-sourceNamespaces|createChannel-specHelmRepo-pathname|createChannel-specHelmRepo-configRef-name|createChannel-specHelmRepo-type|createChannel-configMap-apiVersion|createChannel-configMap-metadata-namespace|createChannel-configMap-metadata-name|createChannel-specObjectBucket-pathname|createChannel-specObjectBucket-secretRef-name|createChannel-specObjectBucket-gate-annotations|createChannel-specGitRepo-pathname|null|_/gi,
    matched => {
      return mapObj[matched]
    }
  )

  return sample
}

export const getSubscriptionSample = locale => {
  const mapObj = {
    'createSubscription-namespace-resource':
      '# ' +
      msgs.get('description.createSubscription.namespace.resource', locale),
    'createSubscription-name-resource':
      '# ' + msgs.get('description.createSubscription.name.resource', locale),
    'createSubscription-apiVersion':
      '# ' + msgs.get('description.createSubscription.apiVersion', locale),
    'createSubscription-kind':
      '# ' + msgs.get('description.createSubscription.kind', locale),
    'createSubscription-metadata-name':
      '# ' + msgs.get('description.createSubscription.metadata.name', locale),
    'createSubscription-metadata-namespace':
      '# ' +
      msgs.get('description.createSubscription.metadata.namespace', locale),
    'createSubscription-spec-channel':
      '# ' + msgs.get('description.createSubscription.spec.channel', locale),
    'createSubscription-spec-placement-placementRef-name':
      '# ' +
      msgs.get(
        'description.createSubscription.spec.placement.placementRef.name',
        locale
      ),
    'createSubscription-spec-placement-placementRef-kind':
      '# ' +
      msgs.get(
        'description.createSubscription.spec.placement.placementRef.kind',
        locale
      ),
    'createSubscription-timeWindow-weekdays':
      '# ' +
      msgs.get('description.createSubscription.timeWindow.weekdays', locale),
    'createSubscription-timeWindow-hours':
      '# ' +
      msgs.get('description.createSubscription.timeWindow.hours', locale),
    null: '',
    _: ' ',
    '\'': ''
  }

  var sample = subscriptionSample.replace(
    /createSubscription-namespace-resource|createSubscription-name-resource|createSubscription-apiVersion|createSubscription-kind|createSubscription-metadata-namespace|createSubscription-metadata-name|createSubscription-spec-channel|createSubscription-spec-placement-placementRef-name|createSubscription-spec-placement-placementRef-kind|createSubscription-timeWindow-weekdays|createSubscription-timeWindow-hours|null|_|'/gi,
    matched => {
      return mapObj[matched]
    }
  )

  return sample
}

export const getApplicationSample = locale => {
  const mapObj = {
    'createApplication-namespace-apiVersion':
      '# ' +
      msgs.get('description.createApplication.namespace.apiVersion', locale),
    'createApplication-namespace-metadata-name':
      '# ' +
      msgs.get('description.createApplication.namespace.metadata.name', locale),
    'createApplication-metadata-name':
      '# ' + msgs.get('description.createApplication.metadata.name', locale),
    'createApplication-metadata-namespace':
      '# ' +
      msgs.get('description.createApplication.metadata.namespace', locale),
    'createApplication-spec-selector-matchExpressions-key':
      '# ' +
      msgs.get(
        'description.createApplication.spec.selector.matchExpressions.key',
        locale
      ),
    'createApplication-spec-selector-matchExpressions-key-values':
      '# ' +
      msgs.get(
        'description.createApplication.spec.selector.matchExpressions.key.values',
        locale
      ),
    null: '',
    _: ' '
  }

  var sample = applicationSample.replace(
    /createApplication-namespace-apiVersion|createApplication-namespace-metadata-name|createApplication-metadata-namespace|createApplication-metadata-name|createApplication-spec-selector-matchExpressions-key-values|createApplication-spec-selector-matchExpressions-key|null|_/gi,
    matched => {
      return mapObj[matched]
    }
  )

  return sample
}

export const getPlacementRuleSample = locale => {
  const mapObj = {
    'createPlacementRule-apiVersion':
      '# ' + msgs.get('description.createPlacementRule.apiVersion', locale),
    'createPlacementRule-kind':
      '# ' + msgs.get('description.createPlacementRule.kind', locale),
    'createPlacementRule-metadata-name':
      '# ' + msgs.get('description.createPlacementRule.metadata.name', locale),
    'createPlacementRule-metadata-namespace':
      '# ' +
      msgs.get('description.createPlacementRule.metadata.namespace', locale),
    'createPlacementRule-spec-clusterSelector-matchLabels':
      '# ' +
      msgs.get(
        'description.createPlacementRule.spec.clusterSelector.matchLabels',
        locale
      ),
    'createPlacementRule-spec-clusterReplicas':
      '# ' +
      msgs.get('description.createPlacementRule.spec.clusterReplicas', locale),
    null: '',
    _: ' '
  }
  var sample = placementRuleSample.replace(
    /createPlacementRule-apiVersion|createPlacementRule-kind|createPlacementRule-metadata-namespace|createPlacementRule-metadata-name|createPlacementRule-spec-clusterSelector-matchLabels|createPlacementRule-spec-clusterReplicas|null|_/gi,
    matched => {
      return mapObj[matched]
    }
  )

  return sample
}
