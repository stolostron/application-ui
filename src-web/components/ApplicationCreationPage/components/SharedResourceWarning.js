// Copyright (c) 2020 Red Hat, Inc. All Rights Reserved.
'use strict'

import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import { ExclamationTriangleIcon } from '@patternfly/react-icons'
import _ from 'lodash'
import apolloClient from '../../../../lib/client/apollo-client'
import { SEARCH_QUERY_RELATED } from '../../../apollo-client/queries/SearchQueries'
import { RESOURCE_TYPES } from '../../../../lib/shared/constants'
import resources from '../../../../lib/shared/resources'
import msgs from '../../../../nls/platform.properties'
import { withLocale } from '../../../providers/LocaleProvider'

resources(() => {
  require('./style.scss')
})

const getSelfLinks = control => {
  const channel = _.get(control, 'groupControlData', []).find(
    d => d.id === 'channel'
  )
  const selfLinks = _.get(channel, 'content', []).find(
    c => c.id === 'selfLinks'
  )
  return _.get(selfLinks, 'active', {})
}

const getResourceKind = resourceType => {
  let kind
  switch (resourceType.name) {
  case 'HCMSubscription':
    kind = 'Subscription'
    break
  case 'HCMPlacementRule':
    kind = 'PlacementRule'
    break
  }
  return kind
}

const getResourceNameAndNamespace = (resourceType, selfLinks) => {
  let selfLinksKey, selfLinkPlural
  switch (resourceType.name) {
  case 'HCMSubscription':
    selfLinksKey = 'Subscription'
    selfLinkPlural = 'subscriptions'
    break
  case 'HCMApplication':
    selfLinksKey = 'Application'
    selfLinkPlural = 'applications'
    break
  case 'HCMPlacementRule':
    selfLinksKey = 'PlacementRule'
    selfLinkPlural = 'placementrules'
    break
  }
  const selfLink = _.get(selfLinks, selfLinksKey, '')
  const selfLinkExp = new RegExp(
    `namespaces/([^/]*)/${selfLinkPlural}/([^/]*)`
  )
  const matches = selfLink.match(selfLinkExp)
  return matches ? matches.slice(1, 3) : [null, null]
}

const getQuery = (resourceType, name, namespace) => {
  const kind = getResourceKind(resourceType).toLowerCase()
  return {
    filters: [
      { property: 'kind', values: [kind] },
      { property: 'name', values: [name] },
      { property: 'namespace', values: [namespace] }
    ],
    relatedKinds: ['application']
  }
}

const SharedResourceWarning = ({ resourceType, control, locale }) => {
  const [relatedApplications, setRelatedApplications] = useState([])
  const selfLinks = getSelfLinks(control)
  const [applicationNamespace, applicationName] = getResourceNameAndNamespace(
    RESOURCE_TYPES.HCM_APPLICATIONS,
    selfLinks
  )
  const [resourceNamespace, resourceName] = getResourceNameAndNamespace(
    resourceType,
    selfLinks
  )

  useEffect(
    () => {
      if (control.editMode && resourceName && resourceNamespace) {
        const query = getQuery(resourceType, resourceName, resourceNamespace)
        apolloClient
          .search(SEARCH_QUERY_RELATED, {
            input: [query]
          })
          .then(response => {
            const relatedItems =
              _.get(response, 'data.searchResult[0].related') || []
            const relatedApps = _.get(
              relatedItems.find(r => r.kind === 'application'),
              'items',
              []
            )
            if (relatedApps) {
              setRelatedApplications(
                relatedApps
                  .filter(
                    r =>
                      r.name !== applicationName ||
                      r.namespace !== applicationNamespace
                  )
                  .map(r => r.name)
                  .sort()
              )
            }
          })
      }
    },
    [
      control.editMode,
      resourceNamespace,
      resourceName,
      applicationName,
      applicationNamespace
    ]
  )

  return relatedApplications.length ? (
    <div className="shared-resource-warning">
      <div>
        <ExclamationTriangleIcon />
      </div>
      <div>
        <p>
          {msgs.get(
            'editing.app.sharedResourceWarning',
            [getResourceKind(resourceType)],
            locale
          )}
        </p>
        <p>{relatedApplications.join(', ')}</p>
      </div>
    </div>
  ) : null
}

SharedResourceWarning.propTypes = {
  control: PropTypes.object,
  locale: PropTypes.string,
  resourceType: PropTypes.object
}

export default withLocale(SharedResourceWarning)
