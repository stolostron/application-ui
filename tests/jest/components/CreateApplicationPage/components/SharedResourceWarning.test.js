// Copyright (c) 2020 Red Hat, Inc.
"use strict";

jest.mock("../../../../../lib/client/apollo-client", () => ({
  search: jest.fn((_, { input: [query] }) => {
    const response = relatedApps => ({
      data: {
        searchResult: [
          {
            related: [
              {
                kind: "application",
                items: [
                  {
                    name: "app-1",
                    namespace: "foo"
                  },
                  ...relatedApps
                ]
              }
            ]
          }
        ]
      }
    });

    const { filters } = query;
    const name = filters.find(f => f.property === "name").values[0];
    let relatedApps;
    switch (name) {
      case "sub-1":
        relatedApps = [
          { name: "shared-sub-app-1" },
          { name: "shared-sub-app-2" },
          { name: "shared-sub-app-3", _hostingSubscription: "bar/sub-1" },
          { name: "app-1", namespace: "bar" }
        ];
        break;
      case "pr-1":
        relatedApps = [
          { name: "shared-pr-app-1" },
          { name: "shared-pr-app-2" }
        ];
        break;
      case "pr-2":
        relatedApps = [];
        break;
    }
    return Promise.resolve(response(relatedApps));
  })
}));

import React from "react";
import { act, create } from "react-test-renderer";
import SharedResourceWarning from "../../../../../src-web/components/ApplicationCreationPage/components/SharedResourceWarning";
import { RESOURCE_TYPES } from "../../../../../lib/shared/constants";

const creationControl = {};

const editingControl = pr => ({
  editMode: true,
  groupControlData: [
    {
      id: "channel",
      content: [
        {
          id: "selfLinks",
          active: {
            Application:
              "/apis/app.k8s.io/v1beta1/namespaces/foo/applications/app-1",
            Subscription:
              "/apis/apps.open-cluster-management.io/v1/namespaces/foo/subscriptions/sub-1",
            PlacementRule: `/apis/apps.open-cluster-management.io/v1/namespaces/foo/placementrules/${pr}`
          }
        }
      ]
    }
  ]
});

let creationMode;
act(() => {
  creationMode = create(
    <SharedResourceWarning
      resourceType={RESOURCE_TYPES.HCM_SUBSCRIPTIONS}
      control={creationControl}
    />
  );
});

let sharedSubscription;
act(() => {
  sharedSubscription = create(
    <SharedResourceWarning
      resourceType={RESOURCE_TYPES.HCM_SUBSCRIPTIONS}
      control={editingControl("pr-1")}
    />
  );
});

let sharedPR;
act(() => {
  sharedPR = create(
    <SharedResourceWarning
      resourceType={RESOURCE_TYPES.HCM_PLACEMENT_RULES}
      control={editingControl("pr-1")}
    />
  );
});

let exclusivePR;
act(() => {
  exclusivePR = create(
    <SharedResourceWarning
      resourceType={RESOURCE_TYPES.HCM_PLACEMENT_RULES}
      control={editingControl("pr-2")}
    />
  );
});

describe("SharedResourceWarning", () => {
  it("renders empty for creation mode", () => {
    expect(creationMode.toJSON()).toBeNull();
  });

  it("renders correctly for a shared subscription", () => {
    expect(sharedSubscription.toJSON()).toMatchSnapshot();
  });

  it("renders correctly for a shared placement rule", () => {
    expect(sharedPR.toJSON()).toMatchSnapshot();
  });

  it("renders correctly for a exclusive placement rule", () => {
    expect(exclusivePR.toJSON()).toBeNull();
  });
});
