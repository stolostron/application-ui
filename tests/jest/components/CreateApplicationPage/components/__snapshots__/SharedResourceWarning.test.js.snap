// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharedResourceWarning renders correctly for a shared placement rule 1`] = `
<div
  className="shared-resource-warning"
>
  <div>
    <svg
      aria-hidden={true}
      aria-labelledby={null}
      fill="currentColor"
      height="1em"
      role="img"
      style={
        Object {
          "verticalAlign": "-0.125em",
        }
      }
      viewBox="0 0 576 512"
      width="1em"
    >
      <path
        d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
        transform=""
      />
    </svg>
  </div>
  <div>
    <p
      dangerouslySetInnerHTML={
        Object {
          "__html": "
              <span class=\\"warning-font\\">Caution: </span>
              This application uses a shared {0} resource. Changes to these settings will also affect the following applications:
              ",
        }
      }
    />
    <p>
      shared-pr-app-1, shared-pr-app-2
    </p>
  </div>
</div>
`;

exports[`SharedResourceWarning renders correctly for a shared subscription 1`] = `
<div
  className="shared-resource-warning"
>
  <div>
    <svg
      aria-hidden={true}
      aria-labelledby={null}
      fill="currentColor"
      height="1em"
      role="img"
      style={
        Object {
          "verticalAlign": "-0.125em",
        }
      }
      viewBox="0 0 576 512"
      width="1em"
    >
      <path
        d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
        transform=""
      />
    </svg>
  </div>
  <div>
    <p
      dangerouslySetInnerHTML={
        Object {
          "__html": "
              <span class=\\"warning-font\\">Caution: </span>
              This application uses a shared {0} resource. Changes to these settings will also affect the following applications:
              ",
        }
      }
    />
    <p>
      shared-sub-app-1, shared-sub-app-2
    </p>
  </div>
</div>
`;
